lint/clang_tidy_wrapper.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/bash
+# This is a wrapper for clang-tidy which allows it to be run on a collection of files
```

Comment:
Could you file and link to an upstream issue about the missing feature upstream? At least that gives us a chance of handing this script to them, and not have to maintain it here.

Reply:
Done, turns out that multiple files are supported, and I missed it. I've now refactored to pass multiple files to clang-tidy. The wrapper is still in the picture at present, distinguishing between errors from checks and fatal errors.

Reply:
I've opened a llvm feature request https://github.com/llvm/llvm-project/issues/95721 to allow distinguishing between fatal and check error through exit code. If that was supported, the wrapper could be removed.